### PR TITLE
xgboost multiclass classification per-class summary

### PR DESCRIPTION
**Pull request by hideaki**
# Description
xgboost multiclass classification per-class summary

# Checklist
Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [x] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/build_xgboost.R [hunk 1]
```diff
@@ -1380,6 +1380,21 @@ tidy.xgboost_exp <- function(x, type = "importance", pretty.name = FALSE, binary
...
+    evaluation_by_class = { # We assume this is called only for multiclass classification.
+      # Delegate showing error for failed models to grance().
+      if ("error" %in% class(x)) {
+        return(glance(x, pretty.name = pretty.name, ...))
+      }
+      # get evaluation scores from training data
+      actual <- extract_actual(x)
+      predicted <- extract_predicted_multiclass_labels(x)
+
+      per_level <- function(class) {
```
Comment (kei51e) [inline R/build_xgboost.R:1392]: It maybe better not to use "class" for the variable name since it is a common function.

Comment (hideaki) [reply R/build_xgboost.R:1392]: I'm now using the name "level" instead.

Comment (kei51e) [inline R/build_xgboost.R:1392]:
```suggestion
      per_level <- function(level) {
```

File: R/build_xgboost.R [hunk 2]
```diff
@@ -1380,6 +1380,21 @@ tidy.xgboost_exp <- function(x, type = "importance", pretty.name = FALSE, binary
...
+        ret <- evaluate_classification(actual, predicted, class, pretty.name = pretty.name)
```
Comment (kei51e) [inline R/build_xgboost.R:1393]:
```suggestion
        ret <- evaluate_classification(actual, predicted, level, pretty.name = pretty.name)
```

**Pull request merged**